### PR TITLE
[ET] Select used et_kernel_metadata only

### PR DESCRIPTION
**Pull request by kirklandsign**
Currently we rely on root operator, but we also need to check for et_kernel_metadata for used specialized kernels.

### CODE REVIEW

File: torchgen/gen_executorch.py [hunk 1]
```diff
@@ -168,8 +168,13 @@ def __call__(
...
 
-        # TODO: Update to use Kernel Selector
-        if not self.selector.is_root_operator(f"{f.namespace}::{f.func.name}"):
+        op_name = f.namespace + "::" + str(f.func.name)
+        if not isinstance(kernel_key, list):
+            kernel_key = [kernel_key]
+        used_kernel_keys = self.selector.et_get_selected_kernels(
+            op_name, [k.to_native_string() for k in kernel_key]
+        )
+        if not used_kernel_keys or not self.selector.is_root_operator(op_name):
```
Comment (larryliu0820) [inline torchgen/gen_executorch.py:177]: Check `is_root_operator()` earlier and skip quickly. Don't need to compute kernel key if not root operator.

File: torchgen/selective_build/selector.py [hunk 2]
```diff
@@ -230,17 +230,34 @@ def is_kernel_dtype_selected(self, kernel_tag: str, dtype: str) -> bool:
...
+                    and key.split("/")[1] == model_kernel_keys.split("/")[1]
+                ):
+                    result_set.add(key)
+                    key_found = True
+                    break
+            if not key_found:
+                if "default" not in kernel_key:
+                    raise Exception("Missing kernel for the model")
+                else:
+                    return ["default"]
```
Comment (larryliu0820) [inline torchgen/selective_build/selector.py:258]: we should add "default" to result_set.

**Review by larryliu0820 — APPROVED** [review]

**Comment by huydhn** [comment]
> "Unrelated error. The BC linter doesn't apply because the API is internal only and is not finalized"

Just FYI, you could add the label `suppress-bc-linter` in the PR to bypass bc linter error.

**Pull request closed without merging**